packages/cli/src/main/kotlin/elide/tool/io/RuntimeWorkdirManager.kt

```diff
@@ -164,23 +172,36 @@ import elide.tool.io.WorkdirManager.WorkdirHandle
     write,
   )
 
-  private fun initializeTemporaryWorkdir(): Path = synchronized(this) {
-    if (initialized) return tempDirectory
-    initialized = true
-    Files.createDirectories(persistentTempPath()).also {
-      rootDirectory = it
+  // private fun initializeTemporaryWorkdir(): Path = synchronized(this) {
+    // if (initialized) return tempDirectory
+    // initialized = true
+    // Files.createDirectories(persistentTempPath()).also {
+      // rootDirectory = it
+    // }
+  // }
+
+  private fun obtainWorkdir(): Path {
+    if (!this::rootDirectory.isInitialized) {
+      rootDirectory = Path.of(projectDirectories.cacheDir)
+      if (!rootDirectory.exists()) {
+        Files.createDirectories(rootDirectory)
+      }
     }
+    return rootDirectory
   }
```

Comment:
your code here improves on what was here, but the approach could use some tuning. i wonder if we can avoid checking the existence / creating the path on first access? maybe we can create a wrapper type for this? ill connect with you on discord about it